packages/interactivity/src/proxies/state.ts

```diff
@@ -277,12 +277,13 @@ const deepMergeRecursive = (
 ) => {
 	if ( isPlainObject( target ) && isPlainObject( source ) ) {
 		for ( const key in source ) {
+			const isNew = ! ( key in target );
```

Comment:
Maybe "hasNewData" or "hasNewProperties" defines better the issue.

Reply:
Here,`isNew` refers to the current `key` inside the loop. But that's a good point anyway; we could move the signal update outside the loop. 🤔

Reply:
Done in https://github.com/WordPress/gutenberg/pull/65135/commits/d516c42db9fa8e081cf93afc5ec60340d2560bf9.